src/main/java/com/targus/experiment/BenchmarkTestRunner.java

```diff
@@ -1,77 +1,282 @@
 package com.targus.experiment;
 
 import com.targus.algorithm.ga.*;
+import com.targus.algorithm.sa.*;
 import com.targus.base.OptimizationProblem;
 import com.targus.base.Solution;
 import com.targus.experiment.wsn.WSNProblemGenerator;
 import com.targus.problem.wsn.WSN;
 import com.targus.problem.wsn.WSNSolutionImprover;
+import com.targus.represent.BitString;
 import com.targus.utils.Constants;
 import javafx.geometry.Point2D;
 
+import java.io.File;
 import java.security.InvalidParameterException;
+import java.util.*;
 
 public class BenchmarkTestRunner {
 
-    public static GA buildGA(OptimizationProblem optimizationProblem, WSN wsn, String gaType, MutationOperator mutationOperator, TerminalState terminalState) {
+    private int seconds;
+    private double GAImprovementRate = Constants.DEFAULT_IMPROVE_PROBABILITY;
+    private double GAImmigrationRate = Constants.DEFAULT_IMMIGRATION_RATE;
+    private int GAPopulationCount = Constants.DEFAULT_POPULATION_COUNT;
+    private String basePath = Constants.DEFAULT_BASE_PATH_FOR_JSON_FILES;
+    private String pathForJsonFiles = basePath + "reference/";
+    private String pathForResults = basePath + "reference_results/";
+
+    public BenchmarkTestRunner(int seconds) {
+        this.seconds = seconds;
+    }
+
+    public BenchmarkTestRunner(int seconds, double GAImprovementRate, double GAImmigrationRate, int GAPopulationCount) {
+        this(seconds);
+        this.GAImprovementRate = GAImprovementRate;
+        this.GAImmigrationRate = GAImmigrationRate;
+        this.GAPopulationCount = GAPopulationCount;
+    }
+
+    private GA buildGA(OptimizationProblem optimizationProblem, WSN wsn, String gaType) {
         GA ga = null;
         switch (gaType) {
             case "Standard":
                 ga = StandardGA.builder(optimizationProblem)
                         .setCrossOverOperator(new OnePointCrossOver())
-                        .setMutationOperator(mutationOperator)
+                        .setMutationOperator(new OneBitMutation())
```

Comment:
OneBitMutation constructor has one parameter which is the mutation rate. The program doesn't run. Am I missing something?